src/software/thunderscope/robot_diagnostics/handheld_device_manager.py

```diff
@@ -0,0 +1,341 @@
+import time
+from logging import Logger
+from threading import Thread, RLock
+
+import numpy
+from evdev import InputDevice, ecodes, list_devices, InputEvent
+from pyqtgraph.Qt import QtCore
+
+import software.python_bindings as tbots_cpp
+from proto.import_all_protos import *
+from software.thunderscope.constants import *
+from software.thunderscope.constants import DiagnosticsConstants
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+from software.thunderscope.robot_diagnostics.handheld_device_status_view import (
+    HandheldDeviceConnectionStatus,
+)
+
+from dataclasses import dataclass
+
+
+@dataclass
+class HandheldDeviceConnectionChangedEvent:
+    connection_status: HandheldDeviceConnectionStatus
+    """The new handheld device connection status"""
+
+    device_name: str | None
+    """The name of the connected handheld device"""
+
+
+class HandheldDeviceManager(QtCore.QObject):
+    """This class is responsible for managing the connection between a computer and a
+    handheld device to control robots. This class relies on the `evdev` python package
+    in order to implement parsing and control flow for device events.
+    See: https://python-evdev.readthedocs.io/en/latest/apidoc.html
+    """
+
+    handheld_device_connection_status_signal = QtCore.pyqtSignal(
+        HandheldDeviceConnectionChangedEvent
+    )
+    """Signal emitted when the handheld device connection status changes"""
+
+    def __init__(
+        self,
+        logger: Logger,
```

Comment:
I don't think it makes much sense to pass `logger` as a parameter into a class. `HandheldDeviceManager` shouldn't depend on `logger`. 

This is what people usually do instead:

`https://stackoverflow.com/questions/18052778/should-a-python-logger-be-passed-as-parameter#:~:text=Not%20usually%3B%20it%20is%20typically,is%20different%20for%20each%20module.`

Are you trying to log to the same file as `DiagnosticsWidget`?

Reply:
People tend to make logger a globla variable I think.

Reply:
Yeah we need to completely redo our logging in python. Boris added these logging changes and I didn't take them out. I'm thinking we should use https://github.com/Delgan/loguru